### PR TITLE
Feat/dlc channels

### PR DESCRIPTION
**Pull request by bonomat**

### CODE REVIEW

File: coordinator/src/node.rs [hunk 1]
```diff
@@ -549,7 +547,9 @@ impl Node {
         let action = if position_contracts + trade_contracts == Decimal::ZERO {
             TradeAction::Close(subchannel.channel_id)
         } else {
-            TradeAction::Resize(subchannel.channel_id)
+            // TODO(bonomat) implement channel resize on dlc-channels
```
Comment (bonomat) [inline coordinator/src/node.rs:550]: https://github.com/get10101/10101/issues/1791

File: coordinator/src/node.rs [hunk 2]
```diff
@@ -731,21 +775,22 @@ impl Node {
...
-        })?;
-        let leverage_coordinator = match channel.liquidity_option_id {
-            Some(liquidity_option_id) => {
-                let liquidity_option = db::liquidity_options::get(&mut conn, liquidity_option_id)?;
-                liquidity_option.coordinator_leverage
-            }
-            None => 1.0,
-        };
+    fn coordinator_leverage_for_trade(&self, _counterparty_peer_id: &PublicKey) -> Result<f32> {
+        // TODO(bonomat): we will need to configure the leverage on the coordinator differently now
```
Comment (bonomat) [inline coordinator/src/node.rs:779]: https://github.com/get10101/10101/issues/1793

File: crates/ln-dlc-node/src/ln_dlc_wallet.rs [hunk 3]
```diff
@@ -278,6 +274,29 @@ impl<S: TenTenOneStorage, N: Storage> dlc_manager::Wallet for LnDlcWallet<S, N>
         _: Option<u64>,
         lock_utxos: bool,
     ) -> Result<Vec<Utxo>, Error> {
+        // TODO(bonomat): this should not be necessary to store it in the db and then load again
```
Comment (bonomat) [inline crates/ln-dlc-node/src/ln_dlc_wallet.rs:277]: https://github.com/get10101/10101/issues/1794

File: crates/ln-dlc-node/src/node/dlc_channel.rs [hunk 4]
```diff
@@ -113,49 +106,50 @@ impl<S: TenTenOneStorage + 'static, N: LnDlcStorage + Sync + Send + 'static> Nod
...
-        let channel_id_hex = hex::encode(channel_id.0);
+    pub fn reject_dlc_channel_offer(&self, channel_id: &DlcChannelId) -> Result<()> {
+        let channel_id_hex = hex::encode(channel_id);
 
         tracing::info!(channel_id = %channel_id_hex, "Rejecting DLC channel offer");
 
-        let (node_id, reject) = self
-            .sub_channel_manager
-            .reject_sub_channel_offer(*channel_id)?;
+        // TODO: implement reject dlc channel offer
```
Comment (bonomat) [inline crates/ln-dlc-node/src/node/dlc_channel.rs:114]: https://github.com/get10101/10101/issues/1795

File: mobile/lib/features/trade/trade_bottom_sheet_tab.dart [hunk 5]
```diff
@@ -321,47 +321,48 @@ class _TradeBottomSheetTabState extends State<TradeBottomSheetTab> {
...
-                  return "Insufficient balance";
-                }
-
-                if (margin.sats > maxMargin) {
-                  return "Max margin is $maxMargin";
-                }
-                if (margin.sats < minTradeMargin.sats) {
-                  return "Min margin is ${minTradeMargin.sats}";
-                }
+                // TODO(bonomat): we need new checks here. For now, YOLO
```
Comment (bonomat) [inline mobile/lib/features/trade/trade_bottom_sheet_tab.dart:324]: https://github.com/get10101/10101/issues/1782

File: mobile/lib/features/wallet/balance.dart [hunk 6]
```diff
@@ -65,14 +65,15 @@ class _BalanceBoxState extends State<_BalanceBox> {
...
   List<String> titles = [
+    'Off-chain',
     'On-chain',
-    'Lightning',
     'USD-P',
   ];
 
   List<BalanceRow> balances = [
-    const BalanceRow(walletType: WalletType.onChain),
+    // TODO(bonomat): this should show the balance from the dlc channel now
```
Comment (bonomat) [inline mobile/lib/features/wallet/balance.dart:74]: https://github.com/get10101/10101/issues/1796

File: mobile/lib/features/wallet/domain/share_payment_request.dart [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import 'package:get_10101/common/domain/model.dart';
 
 class SharePaymentRequest {
+  // TODO(bonomat): this should be removed
```
Comment (bonomat) [inline mobile/lib/features/wallet/domain/share_payment_request.dart:4]: https://github.com/get10101/10101/issues/1796

File: mobile/lib/features/wallet/wallet_screen.dart [hunk 8]
```diff
@@ -46,7 +46,9 @@ class WalletScreen extends StatelessWidget {
                         onPressed: () {
                           context.go((hasChannel || walletChangeNotifier.lightning().sats > 0)
                               ? ReceiveScreen.route
-                              : OnboardingScreen.route);
+                              :
+                              // TODO: we should have a dedicated on-boarding screen for on-boarding with on-chain funds
```
Comment (bonomat) [inline mobile/lib/features/wallet/wallet_screen.dart:50]: https://github.com/get10101/10101/issues/1797

File: mobile/native/src/api.rs [hunk 9]
```diff
@@ -463,23 +463,24 @@ pub struct PaymentRequest {
...
         .map(|amt| format!("?amount={}", Amount::from_sat(amt).to_btc()))
         .unwrap_or_default();
     let addr = ln_dlc::get_unused_address();
 
-    let ln_invoice = if is_usdp {
-        ln_dlc::create_usdp_invoice(amount_sats, description)?
-    } else {
-        ln_dlc::create_invoice(amount_sats, description)?
-    };
+    // TODO(bonomat): this can be removed as we don't support LN for now
```
Comment (bonomat) [inline mobile/native/src/api.rs:474]: https://github.com/get10101/10101/issues/1796

File: mobile/native/src/api.rs [hunk 10]
```diff
@@ -463,23 +463,24 @@ pub struct PaymentRequest {
...
+    // let ln_invoice = if is_usdp {
+    //     ln_dlc::create_usdp_invoice(amount_sats, description)?
+    // } else {
+    //     ln_dlc::create_invoice(amount_sats, description)?
+    // };
 
     Ok(PaymentRequest {
         bip21: format!("bitcoin:{addr}{amount_query}"),
-        lightning: ln_invoice.to_string(),
+        lightning: "TODO: remove this part".to_string(),
```
Comment (bonomat) [inline mobile/native/src/api.rs:483]: https://github.com/get10101/10101/issues/1796

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -193,6 +193,8 @@ jobs:
 
   e2e-tests:
     runs-on: ubuntu-latest
+    # TODO(bonomat): re-run e2e tests
+    if: false
```
Comment (bonomat) [inline .github/workflows/ci.yml:197]: https://github.com/get10101/10101/issues/1798

**Review by 1010Tom — APPROVED** [review]
Looks good to me.

**Comment by bonomat** [comment]
Going ahead and merge this.

**Pull request merged**